.github/workflows/check-docs.yml

```diff
@@ -35,7 +35,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          pip install -e . -U -f https://download.pytorch.org/whl/cpu/torch_stable.html
+          pip install -e ".[dev]" -U --extra-index-url https://download.pytorch.org/whl/cpu
```

Comment:
doctest (almost always) requires optional dependencies in other projects.

Reply:
I may not here use dev, or lat least set is an option as not all our pkgs has these extras and no always you need to install packages like `pytest` for building docs...

Reply:
```suggestion
          pip install -e . -U --extra-index-url https://download.pytorch.org/whl/cpu
```

Reply:
@Borda I understand that `pytest`, for example, isn't necessary for doctest, but we already install `[dev]` requirements in the doctest workflow in PL and Bolts, so I think having it is safe.
- https://github.com/Lightning-AI/lightning/blob/12a061f2aaefaa9ed9ccf81ab6f378835b675a7e/.github/workflows/docs-checks.yml#L52
- https://github.com/Lightning-AI/lightning-bolts/blob/5481b6e8337f4ca1381a9f0d8688f13781ef738e/.github/workflows/docs-check.yml#L32

In 4985dab, I added a new option `pip-install-extras` with the default `[dev]`, but not sure if it'll actually be useful.